app/javascript/components/miq-ae-customization/index.jsx

```diff
@@ -0,0 +1,212 @@
+          onCancel={onCancel}
+          canReset={!!dialogRecord.id}
+          validate={() => {}}
+          FormTemplate={(props) => <FormTemplate {...props} recId={dialogRecord.id} />}
+        />
+      </div>
+    ) : null
+  );
+};
+
+const FormTemplate = ({
+  formFields, recId,
+}) => {
+  const {
+    handleSubmit, onReset, onCancel, getState,
+  } = useFormApi();
+  const { valid, pristine } = getState();
+  const submitLabel = !!recId ? __('Save') : __('Add');
+  return (
+    <form onSubmit={handleSubmit}>
+      {formFields}
+      <FormSpy>
+        {() => (
+          <div className="custom-button-wrapper">
+            { !recId
+              ? (
+                <Button
+                  disabled={!valid}
+                  kind="primary"
+                  className="btnRight"
+                  type="submit"
+                  variant="contained"
+                >
+                  {submitLabel}
+                </Button>
+              ) : (
+                <Button
+                  disabled={!valid || pristine}
+                  kind="primary"
+                  className="btnRight"
+                  type="submit"
+                  variant="contained"
+                >
+                  {submitLabel}
+                </Button>
+              )}
+            {!!recId
+              ? (
+                <Button
+                  disabled={!valid || pristine}
+                  kind="secondary"
+                  className="btnRight"
+                  variant="contained"
+                  onClick={onReset}
+                  type="button"
+                >
+                  { __('Reset')}
+                </Button>
+              ) : null}
+
+            <Button variant="contained" type="button" onClick={onCancel} kind="secondary">
+              { __('Cancel')}
+            </Button>
+          </div>
+        )}
+      </FormSpy>
+    </form>
+  );
+};
+
+MiqAeCustomization.propTypes = {
+  dialogRecord: PropTypes.shape({
+    id: PropTypes.number,
+    name: PropTypes.string,
+    description: PropTypes.string,
+    content: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array]),
+    dialog_type: PropTypes.string,
+  }),
+
+  dialogTypes: PropTypes.arrayOf(
+    PropTypes.arrayOf(
+      PropTypes.string.isRequired
+    ).isRequired
+  ).isRequired,
+};
+
+MiqAeCustomization.defaultProps = {
+  dialogRecord: undefined,
+};
+
+FormTemplate.propTypes = {
+  formFields: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
+    }),
+    PropTypes.shape({
+      name: PropTypes.string,
+    }),
+    PropTypes.shape({
+      description: PropTypes.string,
+    }),
+    PropTypes.shape({
+      content: PropTypes.string,
+    }),
+    PropTypes.shape({
+      dialog_type: PropTypes.string,
+    }),
+  ),
+
+  recId: PropTypes.number,
+};
+
+FormTemplate.defaultProps = {
+  formFields: undefined,
+  recId: undefined,
+};
+
+export default MiqAeCustomization;
```

Comment:
Just a bit of formatting done here to reduce the lines. we don't have to put each props in a new line. we can get a setting in visual studio to auto format it when the number of props are more. if its juts one, put them in a line making it readable.

```
MiqAeCustomization.propTypes = {
  dialogRecord: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string,
    content: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array]),
    dialog_type: PropTypes.string,
  }),
  dialogTypes: PropTypes.arrayOf(
    PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
  ).isRequired,
};

MiqAeCustomization.defaultProps = {
  dialogRecord: undefined,
};

FormTemplate.propTypes = {
  formFields: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.number }),
    PropTypes.shape({ name: PropTypes.string }),
    PropTypes.shape({ description: PropTypes.string }),
    PropTypes.shape({ content: PropTypes.string }),
    PropTypes.shape({ dialog_type: PropTypes.string }),
  ),
  recId: PropTypes.number,
};

FormTemplate.defaultProps = {
  formFields: undefined,
  recId: undefined,
};

export default MiqAeCustomization;

```